src/config_generator/from_proto.rs

```diff
@@ -0,0 +1,441 @@
+#![allow(unused)]
+
+use std::collections::{BTreeSet, HashMap};
+
+use convert_case::{Case, Casing};
+use derive_setters::Setters;
+use prost_reflect::prost_types::{
+    DescriptorProto, EnumDescriptorProto, FileDescriptorSet, ServiceDescriptorProto,
+};
+use strum_macros::Display;
+
+use crate::blueprint::GrpcMethod;
+use crate::config::{Arg, Config, Field, Grpc, Tag, Type};
+
+pub(super) static DEFAULT_SEPARATOR: &str = "__";
+
+/// Enum to represent the type of the descriptor
+#[derive(Display, Clone)]
+enum DescriptorType {
+    Enum,
+    Message,
+    Operation,
+}
+
+impl DescriptorType {
+    fn as_str_name(&self, package: &str, name: &str) -> String {
+        match self {
+            DescriptorType::Enum => {
+                format!("{}{}{}", package, DEFAULT_SEPARATOR, name)
+            }
+            DescriptorType::Message => {
+                format!("{}{}{}", package, DEFAULT_SEPARATOR, name)
+            }
+            DescriptorType::Operation => format!(
```

Comment:
We should consider building nested operations for handling namespaces. So for example if the method is `a.b.c.GetNews` the equivalent GraphQL query should be `query {a { b {c { getNews } } }}`